### PR TITLE
bpo-36504: Fix signed integer overflow in _ctypes.c's PyCArrayType_new()

### PR DESCRIPTION
**Pull request by ZackerySpytz**
https://bugs.python.org/issue36504

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by ZackerySpytz** [comment]
Hi, @serhiy-storchaka. It seems that the bot failed to create the backports?

**Comment by serhiy-storchaka** [comment]
Do you mind to create a backport to 2.7 manually if the bug exists in 2.7?

**Comment by ZackerySpytz** [comment]
I created the backport to 2.7.

File: Modules/_ctypes/_ctypes.c [hunk 1]
```diff
@@ -1518,7 +1518,7 @@ PyCArrayType_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     }
 
     itemsize = itemdict->size;
-    if (length * itemsize < 0) {
+    if (length > PY_SSIZE_T_MAX / itemsize) {
```
Comment (eric-wieser) [inline Modules/_ctypes/_ctypes.c:1521]: Divide-by-zero for:
```
class S(ctypes.Structure): _fields_ = []

(S*10)()
```

[bpo13881](https://bugs.python.org/issue37188)

**Pull request merged**